### ISSUE TITLE
DestinationRule caCertificates system option is not documented

### ISSUE THREAD

**Issue by bh-tt** [istio/istio#49783]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

https://github.com/istio/istio/pull/33472#discussion_r706477304 says one can use the `system` option for DestinationRule `/spec/trafficPolicy/tls/caCertificates`, but this is not documented on the istio [docs](https://istio.io/latest/docs/reference/config/networking/destination-rule/#ClientTLSSettings) website (and I'm not sure the previous comment about the webhook validation is implemented). 

Documentation should be added to help users find what cacerts are being used here.

### Version

```Text
Not really relevant here, as it pertains to the docs
```

### Additional Information

Not sure if I should submit this here or on the site source repo?

**Comment by leosarra** [comment]
Hello @bh-tt ,
Sorry for the late reply but I forgot to link the issue to the PR.
2 weeks ago I updated the documentation regarding certificates https://github.com/istio/api/pull/3122
I opted to not document "system" as an option as, from my understanding, it was not expected to become one, hence why the reviewer was originally pushing to have the validation logic that prevented users from using inside DR.
On the other end the documentation of the certificate required multiple improvements which I took care of:
- document that the default behaviour consists now of checking the certificate against the OS CA (same as what you would have if you were to set "system").
- update documentation for disabling the OS CA check and remove mentions of deprecated ENV variables
